contracts/optimistic-ethereum/OVM/bridge/tokens/Abs_L1TokenGateway.sol

```diff
@@ -0,0 +1,212 @@
+     * In most cases, this will simply send locked funds to the withdrawer.
+     *
+     * @param _to Address being withdrawn to.
+     * @param _amount Amount being withdrawn.
+     */
+
+    function _handleFinalizeWithdrawal(
+        address _to,
+        uint256 _amount
+    )
+        internal
+        virtual
+    {
+        revert("Implement me in child contracts");
+    }
+
+    /**
+     * @dev Core logic to be performed when a deposit is initiated on L1.
+     * In most cases, this will simply send locked funds to the withdrawer.
+     *
+     * @param _from Address being deposited from on L1.
+     * @param _to Address being deposited into on L2.
+     * @param _amount Amount being deposited.
+     */
+
+    function _handleInitiateDeposit(
+        address _from,
+        address _to,
+        uint256 _amount
+    )
+        internal
+        virtual
+    {
+        revert("Implement me in child contracts");
+    }
+
+    /**
+     * @dev Overridable getter for the L2 gas limit, in the case it may be
+     * dynamic, and the above public constant does not suffice.
+     *
+     */
+
+    function getFinalizeDepositL2Gas()
+        public
+        view
+        override
+        returns(
+            uint32
+        )
+    {
+        return DEFAULT_FINALIZE_DEPOSIT_L2_GAS;
+    }
+
+    /**************
+     * Depositing *
+     **************/
+
+    /**
+     * @dev deposit an amount of the ERC20 to the caller's balance on L2
+     * @param _amount Amount of the ERC20 to deposit
+     */
+    function deposit(
+        uint _amount
+    )
+        public
+        override
+    {
+        _initiateDeposit(msg.sender, msg.sender, _amount);
+    }
+
+    /**
+     * @dev deposit an amount of ERC20 to a recipients's balance on L2
+     * @param _to L2 address to credit the withdrawal to
+     * @param _amount Amount of the ERC20 to deposit
+     */
+    function depositTo(
+        address _to,
+        uint _amount
+    )
+        public
+        override
+    {
+        _initiateDeposit(msg.sender, _to, _amount);
+    }
+
+    /**
+     * @dev Performs the logic for deposits by informing the L2 Deposited Token
+     * contract of the deposit and calling a handler to lock the L1 funds. (e.g. transferFrom)
+     *
+     * @param _from Account to pull the deposit from on L1
+     * @param _to Account to give the deposit to on L2
+     * @param _amount Amount of the ERC20 to deposit.
+     */
+    function _initiateDeposit(
+        address _from,
+        address _to,
+        uint _amount
+    )
+        internal
+    {
+        // Call our deposit accounting handler implemented by child contracts.
+        _handleInitiateDeposit(
+            _from,
+            _to,
+            _amount
+        );
+
+        // Construct calldata for l2DepositedToken.finalizeDeposit(_to, _amount)
+        bytes memory data = abi.encodeWithSelector(
+            iOVM_L2DepositedToken.finalizeDeposit.selector,
```

Comment:
Whoa! When did Solidity add this?

Reply:
I think it's been there for a bit, but it still lets you pass in the wrong argument types sooo meh lol